keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/unary/ArcCosVertex.java

```diff
@@ -0,0 +1,30 @@
+package io.improbable.keanu.vertices.dbl.nonprobabilistic.operators.unary;
+
+import io.improbable.keanu.vertices.dbl.DoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.ConstantDoubleVertex;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.diff.DualNumber;
+import io.improbable.keanu.vertices.dbl.nonprobabilistic.diff.Infinitesimal;
+
+public class ArcCosVertex extends DoubleUnaryOpVertex {
+
+    public ArcCosVertex(DoubleVertex inputVertex) {
+        super(inputVertex);
+    }
+
+    public ArcCosVertex(double inputValue) {
+        super(new ConstantDoubleVertex(inputValue));
+    }
+
+    @Override
+    protected Double op(Double a) {
+        return Math.acos(a);
+    }
+
+    @Override
+    public DualNumber getDualNumber() {
+        DualNumber inputDualNumber = inputVertex.getDualNumber();
+        double dArcCos = Math.PI / 2 - Math.asin(inputVertex.getValue());
+        Infinitesimal outputInfinitesimal = inputDualNumber.getInfinitesimal().multiplyBy(dArcCos);
+        return new DualNumber(Math.cos(inputVertex.getValue()), outputInfinitesimal);
```

Comment:
For the value of the dual number, use the op method. That way we don't have to test the dual number value is correct if we've already tested the op method.

new DualNumber(Math.cos(inputVertex.getValue()), outputInfinitesimal) -> 
new DualNumber(op(inputVertex.getValue()), outputInfinitesimal)

Reply:
Good spot - updated this across the board